src/main/java/bio/terra/service/dataset/StorageResourceDao.java

```diff
@@ -95,7 +95,13 @@ public void createStorageAttributes(List<StorageResource> storageResources, UUID
             String platformParam = storageResource.getCloudResource() + "_cloudPlatform";
             valuesList.add(String.format("(:dataset_id, :%s, :%s, :%s)",
                 regionParam, cloudResourceParam, platformParam));
-            params.addValue(regionParam, storageResource.getRegion().name());
+
+            if (storageResource.getCloudResource() == GoogleCloudResource.FIRESTORE) {
+                params.addValue(regionParam, storageResource.getRegion().getFirestoreFallbackRegion().name());
```

Comment:
Is this special case handling a concern for `StorageResourceDao` or for `StorageResource`? It could be done inside `StorageResource`, since it has all the information needed to make this decision, but I'm not sure if the logic belong there or not.

Reply:
I'm not sure. I see `StorageResource` as just a container bean, whereas logic would belong in `StorageResourceDao`. I could be convinced the other way though.